keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+  - [Scalability](#scalability)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [NetworkPolicy v2](#networkpolicy-v2)
+  - [Empower, Deny, Allow action based CRD](#empower-deny-allow-action-based-crd)
+    - [ClusterDefaultNetworkPolicy resource](#clusterdefaultnetworkpolicy-resource)
+  - [Single CRD with DefaultRules field](#single-crd-with-defaultrules-field)
+  - [Single CRD with IsOverrideable field](#single-crd-with-isoverrideable-field)
+  - [Single CRD with BaselineAllow as Action](#single-crd-with-baselineallow-as-action)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Introduce new set of APIs to express an administrator's intent in securing
+their K8s cluster. This doc proposes the AdminNetworkPolicy API to complement
+the developer focused NetworkPolicy API in Kubernetes.
+
+## Motivation
+
+Kubernetes provides the NetworkPolicy resource to control traffic within a
+cluster. NetworkPolicy focuses on expressing a developer's intent to secure
+their applications. However, it was not intended to be used for cluster scoped
+administrative traffic control, which is reflected by its design:
+- NetworkPolicy uses a "implicit isolation" model, which means that once a policy
+  is applied to certain workloads, they are automatically isolated (in the direction
+  specified by the policy) and anything allowed needs to be explicitly called out.
+  It has no concept of explicit "deny" rules, because the application deployer can
+  simply refrain from allowing the things they want to deny.
+- It doesn't include priorities, or any concept of one user "outranking" another
```

Comment:
Nit: this sentence misses the point of NetworkPolicy. NP was specifically designed to be commutative (in other words, ordering doesn't matter). Perhaps something like "The commutative nature of NetworkPolicy can make certain filtering intents difficult to express."

Reply:
Ack I agree, will update

Reply:
done